config/configuration.go

```diff
@@ -320,9 +320,14 @@ func (c *configuration) GetEthereumMaxRetries() int {
 	return c.GetInt("ethereum.maxRetries")
 }
 
-// GetEthereumGasPrice returns the gas price to use for a ethereum transaction.
-func (c *configuration) GetEthereumGasPrice() *big.Int {
-	return big.NewInt(cast.ToInt64(c.get("ethereum.gasPrice")))
+// GetEthereumMaxGasPrice returns the gas price to use for a ethereum transaction.
+func (c *configuration) GetEthereumMaxGasPrice() *big.Int {
+	n := new(big.Int)
+	n, ok := n.SetString(c.GetString("ethereum.maxGasPrice"), 10)
+	if !ok {
+		log.Error("could not read ethereum.maxGasPrice")
```

Comment:
Is 0 the default value if not set in config?

Reply:
its just nil, so it will cause a nil pointer error and node will crash. I guess thats better than continuing to run since its misconfigured.

Reply:
I think I will make this a log.Fatal 
what do you think?